enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,754 @@
+When the response is denied, the reconcile should be aborted.
+
+If the webhook is unavailable, the mechanism should fail close and the reconcile should be retried at a later time.
+
+#### Filtering reconciles in Machine API controllers
+
+Each controller will, at the start of the reconcile logic, check for the presence of the authoritative annotation and will either skip or continue with the reconciliation based on the presence of this annotation.
+
+#### The Cluster Autoscaler in mixed MachineSet clusters
+
+The Cluster Autoscaler today looks at Machine API MachineSets and filters these based on a number of annotations to identify node groups that it may scale.
+In the future, where we have both Machine API and Cluster API authoritative MachineSets, the Autoscaler must be updated to ensure it scales the authoritative MachineSet.
+
+Since every Machine API MachineSet will be reflected into Cluster API, the autoscaler can be updated to reconcile only the Cluster API MachineSets.
+When a scaling decision is made, it must then update the authoritative MachineSet to the correct scale.
+
+To do this, there are two possible approaches to consider:
+- Look up the Machine API MachineSet mirror to determine the authoritative API and update accordingly
+  - This would be a carry patch to the autoscaler, which, may cause maintenance toil over time
+- Allow the Autoscaler to update the Cluster API MachineSet, but allow the synchronisation controller to identify the update has come from the Autoscaler, and mirror it to the authoritative MachineSet as appropriate
+  - This could be achieved by observing the managed fields property of the replicas field and determine which actor updated it.
+  If the Autoscaler last changed the replica count, it should be mirrored back to the Machine API MachineSet if it is authoritative.
+  - This creates an exception to the rules outline above and adds complication to the conversion, but, may be less of a burden than carrying a patch in the autoscaler
+
+Both avenues above will need to be explored further before a decision is made on how the Autoscaler will handle mixed MachineSet clusters.
+
+#### Validation of the authoritative API
+
+The authoritative API has two valid values, `machine.openshift.io` and `cluster.x-k8s.io`. Any other value is an error.
+To prevent invalid values, a new validating admission webhook will be introduced to validate the value of the annotation.
+
+This webhook will be configured to accept resources from the Machine API group and check the annotation value is permitted.
+Any invalid value will be rejected by the webhook.
+
+In addition to this, the webhook will validate, on a change of authoritative API, that the synchronisation is up to date.
+If the value of the `machine.openshift.io/synchronised-generation` annotation does not match the current generation,
+the change will be rejected and requested to be retried once the synchronisation has occurred.
```

Comment:
The intention is this only is checked when there's a change of the authoritative API, as in, if you change from MAPI to CAPI or vice versa, so I don't think it will actually cause issues in most cases.

However, I think if we move to the controller "accepting" a change to the authoritative API, that we will be able to drop this anyway